[GLib] Expand GI_DOCGEN templates through unifdef

aperezdc opened the pull request:
#### 25abf03a5d0d2af6b48426e15eadd1c26104cde2
<pre>
[GLib] Expand GI_DOCGEN templates through unifdef
<a href="https://bugs.webkit.org/show_bug.cgi?id=283192">https://bugs.webkit.org/show_bug.cgi?id=283192</a>

Reviewed by Michael Catanzaro.

Pass templates listed in the CONTENT_TEMPLATES parameter to the
GI_DOCGEN() function through unifdef, to allow more sharing of
documentation sources among the GTK and WPE ports.

* Source/WebKit/PlatformGTK.cmake: List environment-variables.md as a
  documentation template file.
* Source/WebKit/PlatformWPE.cmake: Ditto.
* Source/WebKit/glib/environment-variables.md.in: Renamed
  from Source/WebKit/wpe/environment-variables.md, slightly edited to
  make the introduction generic, and guarded the WPE-specific section.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in: List
  environment-variables.md for inclusion in the documentation.
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in: Ditto.
* Source/cmake/FindGIDocgen.cmake: Arrange to pass content template
  files through unifdef. The defined macros are GTK, GTK3, GTK4, and
  WPE.

Canonical link: <a href="https://commits.webkit.org/286762@main">https://commits.webkit.org/286762@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6306e9af792d20715a99c93b6c4775b7727d5294

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77031 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/56066 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29946 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/81581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/28310 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/65214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4362 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/81581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/28310 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80098 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/65214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/66123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/81581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/65214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/26636 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/70216 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/65214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/23949 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/83015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/76309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/4411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/2959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/83015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/4566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/66096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/83015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9961 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/98562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11915 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/4357 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/7173 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/98562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/4377 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/7812 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/6136 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/cmake/FindGIDocgen.cmake:
```diff
@@ -60,6 +60,7 @@ if (NOT ENABLE_DOCUMENTATION)
 endif ()
 
 find_package(GI REQUIRED)
+find_package(Unifdef REQUIRED)
```
Please make sure this still works when building with `-DUSE_SYSTEM_UNIFDEF=OFF`?

aperezdc replied:
Good point, I'll make sure before landing.

aperezdc replied:
So, this line is not needed, because we already have this in `Source/CMakeLists.txt`:

```cmake
if (USE_SYSTEM_UNIFDEF)
  find_package(Unifdef REQUIRED)
else ()
  add_subdirectory(ThirdParty/unifdef)
endif ()
```

So I'll remove it before landing.

(Now, a different discussion is whether we should fix the above to avoid checking for `unifdef` or building it for ports other than WPE/GTK. That would belong in another patch.)

mcatanzaro reviewed (approved):
I'm going to approve this on the assumption that you are planning to follow up by documenting more environment variables, since it doesn't make sense to do this if 100% of the documented environment variables are compiled out when building for GTK. Presumably you wouldn't do this without plans for a follow-up.

P.S. I'm not actually convinced it's a good idea to document environment variables, since the documentation is likely to become stale when environment variables are added or removed -- we probably have hundreds of environment variables! -- and since the environment variables are probably not very useful for developers. But using unifdef to preprocess the documentation is a great idea.

aperezdc committed:
[GLib] Expand GI_DOCGEN templates through unifdef
https://bugs.webkit.org/show_bug.cgi?id=283192

Reviewed by Michael Catanzaro.

Pass templates listed in the CONTENT_TEMPLATES parameter to the
GI_DOCGEN() function through unifdef, to allow more sharing of
documentation sources among the GTK and WPE ports.

* Source/WebKit/PlatformGTK.cmake: List environment-variables.md as a
  documentation template file.
* Source/WebKit/PlatformWPE.cmake: Ditto.
* Source/WebKit/glib/environment-variables.md.in: Renamed
  from Source/WebKit/wpe/environment-variables.md, slightly edited to
  make the introduction generic, and guarded the WPE-specific section.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in: List
  environment-variables.md for inclusion in the documentation.
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in: Ditto.
* Source/cmake/FindGIDocgen.cmake: Arrange to pass content template
  files through unifdef. The defined macros are GTK, GTK3, GTK4, and
  WPE.

Canonical link: https://commits.webkit.org/286762@main
